src/utils/dummy_generator.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+   Part of the Battle for Wesnoth Project http://www.wesnoth.org/
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+   See the COPYING file for more details.
+*/
+
+/**
+ *  @file
+ *  Fallback for name generation (to guarantee, that name_generator_factory always returns a valid object reference)
```

Comment:
This leads to the following question: Why does `name_generator_factory` always need to return a valid object reference? It could easily return a pointer or a `boost::optional` instead, for example.